storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
Yeah in line 541 we are assigning over the `err` declared in the signature.
Let's make that clearer too.